src/core/lib/gprpp/fork.cc

```diff
@@ -63,6 +64,13 @@ class ExecCtxState {
   }
 
   void IncExecCtxCount() {
+    // If we are in an EventEngine thread, we should not block any ExecCtx
```

Comment:
I don't think this is sound: event engine threads can ultimately end up executing anything that schedules on an `ExecCtx`, so I don't buy the claim that we don't need to worry about them.

Reply:
I might be missing something here -- not an expert on EventEngine -- but I think the issue is just that the wording of the comment here is just slightly off. It's not that we're not worrying about them  it's that the ThreadPool implementation is expected to deal with the pre-fork behavior rather than this method.

Reply:
Richard is right. The wording is loose, the behavior is not.

Reply:
Can we update this comment with more precise language then? I assume it's "EventEngine is expected to terminate all threads before fork, and so this extra work is unnecessary" or something along those lines?

Reply:
Done